internal/consumer/partitionConsumer.go

```diff
@@ -404,6 +413,27 @@ func (p *rangePartitionConsumer) ResetOffset(offsetRange kafka.OffsetRange) erro
 		return errors.New("rangePartitionConsumer expects a highwatermark when resetting offset")
 	}
 
+	// If it has been closed, return error.
+	select {
+	case _, ok := <-p.stopC:
+		if !ok {
+			return errors.New("unable to ResetOffset for rangePartitionConsumer has been closed")
+		}
+	case <-time.After(time.Nanosecond):
```

Comment:
do we really need the select case here?

Reply:
The select with timer is necessary to ensure no deadlock  on `_, ok := <-ch`.

Reply:
Isn't a default sufficient?